src/Setup/UpdateVersions.purs

```diff
@@ -92,7 +92,11 @@ fetchLatestReleaseVersion tool = Tool.repository tool # case tool of
                 [ "Failed to decode tag from GitHub response: ", parseErrorMessage e ]
 
             Right v ->
-              pure v
+              if Version.isPreRelease v then
+                throwError $ error $ fold
+                  [ "Latest version is prerelease version: ", show v ]
```

Comment:
Not sure if throwing an error is the appropriate action here.

Reply:
Hm. I don’t think we want an exception because then none of the other tools will be updated if any tool is on a prerelease version.

Perhaps we could either:

1. Read the versions file and fall back to its contents for this tool if the latest version is a prerelease
2. Fetch more of the releases to find the first one which is not a pre-release
3. Have these functions return a Maybe, and rather than always overwrite the versions file, only modify it with the fields for tools that succeeded (ie. weren’t prerelease versions)

Reply:
#1 would mean reading the old versions Json file in this repository (which could be foreign-imported, I believe) to get all the fallback releases.

#2 would mean fetching the last N releases instead of only the latest release, and then iterating over them to find the latest non-pre-release version. I’m not sure what GitHub API call to use but there must be one.

#3 would also mean reading in the versions file and then modifying that Json only with the tools that found a non-pre-release version.

Reply:
I think ideally we'd choose 2, but we'll have to find a GitHub API call that returns all the releases of a repository as you stated. I'll research this a little bit, and if I can't find anything, we'll fall back to 1 or 3.

Reply:
I found something: https://docs.github.com/en/rest/reference/repos#releases. I think the `per_page` parameter is what we want. I guess if none of the N releases we get are not pre-releases, we will move to the next page using the `page` parameter. Do you see anything wrong with that?

Reply:
I think for a sufficiently large N (say 15?) we can just throw the exception if none of them are valid. After all, these tools don’t generally even do pre-releases, and if they do it’s normally just one.

Reply:
I see your point, but I think it's better to be future proof. The tools could change their versioning model at any time. I also don't think it's too difficult to check all releases.

Reply:
Yea, I’m fine with that if it’s not too much work for you!

Reply:
It's not too much for me, so don't worry!

Reply:
Uh oh 😅 https://github.com/purescript/purescript/releases/tag/v0.14.0-rc.1

I'm going to disable checking for latest versions until this PR is in.

Reply:
Do you mind reverting this as part of this PR?
https://github.com/purescript-contrib/setup-purescript/commit/d2ad9a714b9063b575ac323fabd0466c3db1b8b7

Reply:
Will do!